docs/interactions/Slash_Commands.md

```diff
@@ -0,0 +1,841 @@
+| description | string                                                                                                         | 1-100 character description                                                                                |
+| default?    | bool                                                                                                           | the first `required` option for the user to complete--only one option can be `default`                     |
+| required?   | bool                                                                                                           | if the parameter is required or optional--default `false`                                                  |
+| choices?    | array of [ApplicationCommandOptionChoice](#DOCS_INTERACTIONS_SLASH_COMMANDS/applicationcommandoptionchoice) | choices for `string` and `int` types for the user to pick from                                             |
+| options?    | array of [ApplicationCommandOption](#DOCS_INTERACTIONS_SLASH_COMMANDS/applicationcommandoption)              | if the option is a subcommand or subcommand group type, this nested options will be the parameters         |
+
+## ApplicationCommandOptionType
+
+| Name              | Value |
+|-------------------|-------|
+| SUB_COMMAND       | 1     |
+| SUB_COMMAND_GROUP | 2     |
+| STRING            | 3     |
+| INTEGER           | 4     |
+| BOOLEAN           | 5     |
+| USER              | 6     |
+| CHANNEL           | 7     |
+| ROLE              | 8     |
+
+## ApplicationCommandOptionChoice
+
+If you specify `choices` for an option, they are the **only** valid values for a user to pick
+
+| Field | Type          | Description         |
+|-------|---------------|---------------------|
+| name  | string        | 1-100 character choice name |
+| value | string or int | value of the choice |
+
+## Interaction
+
+An interaction is the base "thing" that is sent when a user invokes a command, and is the same for Slash Commands and other future interaction types
+
+| Field      | Type                              | Description                                             |
+|------------|-----------------------------------|---------------------------------------------------------|
+| id         | snowflake                         | id of the interaction                                   |
+| type       | InteractionType                   | the type of interaction                                 |
+| data?\*    | ApplicationCommandInteractionData | the command data payload                                |
+| guild_id   | snowflake                         | the guild it was sent from                              |
+| channel_id | snowflake                         | the channel it was sent from                            |
+| member     | GuildMember                       | guild member data for the invoking user                 |
```

Comment:
Is this guaranteed to have the User attached, unlike `CREATE_MESSAGE`?

Reply:
Yes the user object is guaranteed